backend/src/transactions/club.go

```diff
@@ -38,8 +40,18 @@ func GetClubs(db *gorm.DB, queryParams *models.ClubQueryParams) ([]models.Club,
 
 	if queryParams.Tags != nil && len(queryParams.Tags) > 0 {
 		query = query.Joins("JOIN club_tags ON club_tags.club_id = clubs.id").
-			Where("club_tags.tag_id IN ?", queryParams.Tags).
-			Group("clubs.id") // ensure unique club records
+			Where("club_tags.tag_id IN ?", queryParams.Tags). // add search function here
+			Group("clubs.id")                                 // ensure unique club records
+	}
+
+	if queryParams.Search != "" {
```

Comment:
what if we make queryParams.Search a *string to make this a nil check vs a default value check? does empty string here represent the absence of a value or an empty search query?